### PR TITLE
bpo-34666: Implement stream.awrite() and stream.aclose()

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue34666

### CODE REVIEW

**Comment by asvetlov** [comment]
@willingc please review documentation changes.

File: Doc/library/asyncio-stream.rst [hunk 1]
```diff
@@ -229,14 +229,57 @@ StreamWriter
...
+   .. method:: get_extra_info(name, default=None)
+
+      Access optional transport information; see
+      :meth:`BaseTransport.get_extra_info` for details.
+
    .. method:: write(data)
 
       Write *data* to the stream.
 
+      The method doesn't apply control-flow, the call should be
```
Comment (willingc) [inline Doc/library/asyncio-stream.rst]: This method doesn't apply control-flow. The call should be followed by...

File: Doc/library/asyncio-stream.rst [hunk 2]
```diff
@@ -229,14 +229,57 @@ StreamWriter
...
+      followed by :meth:`drain`.
+
    .. method:: writelines(data)
 
       Write a list (or any iterable) of bytes to the stream.
 
+      The method doesn't apply control-flow, the call should be
```
Comment (willingc) [inline Doc/library/asyncio-stream.rst]: Same correction as above.

File: Misc/NEWS.d/next/Library/2018-09-13-11-49-52.bpo-34666.3uLtWv.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Implement ``asyncio.StreamWriter.awrite`` and
+``asyncio.StreamWriter.aclose()`` coroutines.  Methods are needed for
+providing consistent stream API with control flow switched on by default.
```
Comment (willingc) [inline Misc/NEWS.d/next/Library/2018-09-13-11-49-52.bpo-34666.3uLtWv.rst]: providing a consistent

**Review by willingc — COMMENTED** [review]
Looks great. A few tiny typos.

File: Doc/library/asyncio-stream.rst (hunk 1 above)
Comment (asvetlov) [reply Doc/library/asyncio-stream.rst]: done

File: Doc/library/asyncio-stream.rst (hunk 2 above)
Comment (asvetlov) [reply Doc/library/asyncio-stream.rst]: done

File: Misc/NEWS.d/next/Library/2018-09-13-11-49-52.bpo-34666.3uLtWv.rst (hunk 3 above)
Comment (asvetlov) [reply Misc/NEWS.d/next/Library/2018-09-13-11-49-52.bpo-34666.3uLtWv.rst]: done

**Review by willingc — APPROVED** [review]
Thanks @asvetlov.

**Comment by asvetlov** [comment]
@willingc thanks for the review!

File: Doc/library/asyncio-stream.rst [hunk 4]
```diff
@@ -229,14 +229,57 @@ StreamWriter
...
+
+      Return *True* if the underlying transport supports
+      the :meth:`write_eof` method, *False* otherwise.
+
+   .. method:: write_eof()
+
+      Close the write end of the stream after the buffered write
+      data is flushed.
+
+   .. attribute:: transport
```
Comment (1st1) [inline Doc/library/asyncio-stream.rst:260]: I actually moved these sections down on purpose...

**Pull request merged**